WordPress/gutenberg
# RangeControl: Convert component to TypeScript

Depends On:
- https://github.com/WordPress/gutenberg/pull/40518

Related:
- https://github.com/WordPress/gutenberg/issues/40507
- https://github.com/WordPress/gutenberg/pull/41444
- https://github.com/WordPress/gutenberg/pull/41445

## What?
Converts the `RangeControl` component to TypeScript. 

_(Unit tests and story TypeScript conversions are in separate PRs, #41444 & #41445)_

## Why?
While looking to create a new component combining both a `UnitControl` and `RangeControl` (https://github.com/WordPress/gutenberg/pull/40462) it presented an opportunity to improve the slider components we offered in a more composable way. 

The first step in this process was to type the `RangeControl` component.

See https://github.com/WordPress/gutenberg/issues/40507 for more details.

## How?
Refactors the current `RangeControl` to TypeScript.

## Testing Instructions
1. Run `npm run dev:package-types` and ensure no typing errors
2. Run `npm run test-unit packages/components/src/range-control/test/` and check they all pass
3. Fire up the Storybook and confirm the RangeControl continues to function as expected
4. Test the various range controls within the block/site editors in Gutenberg still work
